Public/Src/Cache/DistributedCache.Host/Service/Internal/DualServerClientContentStore.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.Threading;
+using System.Threading.Tasks;
+using BuildXL.Cache.ContentStore.Distributed.Redis.Credentials;
+using BuildXL.Cache.ContentStore.Interfaces.Distributed;
+using BuildXL.Cache.ContentStore.Interfaces.FileSystem;
+using BuildXL.Cache.ContentStore.Interfaces.Logging;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Interfaces.Sessions;
+using BuildXL.Cache.ContentStore.Interfaces.Stores;
+using BuildXL.Cache.ContentStore.Interfaces.Tracing;
+using BuildXL.Cache.ContentStore.Service;
+using BuildXL.Cache.ContentStore.Sessions;
+using BuildXL.Cache.ContentStore.Stores;
+using BuildXL.Cache.Host.Configuration;
+
+namespace BuildXL.Cache.Host.Service.Internal
+{
+    /// <summary>
+    /// Content store containing both a distributed <see cref="LocalContentStore"/> and <see cref="ServiceClientContentStore"/>.

```

Comment:
The comment tells about the inner structure of the class but doesn't explain why we need this inner structure.

Could you please explain why we have to combine service client and localcontentserver in one place? I think it should be valuable.